scripts/build/Jenkins/Jenkinsfile

```diff
@@ -433,26 +429,35 @@ def ExportTestScreenshots(Map options, String branchName, String platformName, S
     }
 }
 
-def UploadAPLogs(Map options, String branchName, String platformName, String jobName, String workspace, Map params) {
+// All files are included by default.
+// --include will only re-include files that have been excluded from an --exclude filter.
+//See more details at https://docs.aws.amazon.com/cli/latest/reference/s3/#use-of-exclude-and-include-filters
+def ArchiveArtifactsOnS3(String artifactsSourceDir, String s3Prefix="", boolean recursive=false, List<String> includes=[], List<String> excludes=[]) {
+    if (!fileExists(s3Prefix)) {
+        palMkdir(s3Prefix)
+    }
+    palSh("echo ${env.BUILD_URL} > ${s3Prefix}/build_url.txt")
+    // archiveArtifacts is very slow, so we only archive one file and upload the rest artifacts to the same bucket using S3 CLI.
+    archiveArtifacts artifacts: "${s3Prefix}/build_url.txt"
+    def command = "aws s3 cp ${artifactsSourceDir} s3://${env.JENKINS_ARTIFACTS_S3_BUCKET}/${env.JENKINS_JOB_NAME}/${env.BUILD_NUMBER}/artifacts/${s3Prefix} "
+    includes.each{ include ->
+        command += "--include \"${include}\" "
+    }
+    excludes.each{ exclude ->
+        command += "--exclude \"${exclude}\" "
+    }
+    if (recursive) command += "--recursive "
```

Comment:
Almost forgot: You still need to add a `--acl bucket-owner-full-control` argument to `aws s3 cp` to support cross account permissions on the object

Reply:
Discussed this offline. This is all within the same AWS account, so cross account won't be happening. Dropping the issue.